controllers/barbican_controller.go

```diff
@@ -661,6 +663,25 @@ func (r *BarbicanReconciler) generateServiceConfig(
 		"EnableSecureRBAC": instance.Spec.BarbicanAPI.EnableSecureRBAC,
 	}
 
+	templateParameters["UseApplicationCredentials"] = false
+	// fetch AC CR
+	ac := &keystonev1.ApplicationCredential{}
+	acName := types.NamespacedName{Namespace: instance.Namespace, Name: fmt.Sprintf("ac-%s", barbican.ServiceName)}
+	if err := r.Client.Get(ctx, acName, ac); err == nil {
+		// fetch AC Secret
+		secret := &corev1.Secret{}
+		secName := types.NamespacedName{Namespace: ac.Namespace, Name: ac.Status.SecretName}
+		if err := r.Client.Get(ctx, secName, secret); err == nil {
+			// switch to application credentials auth
+			templateParameters["UseApplicationCredentials"] = true
+			templateParameters["ACID"] = string(secret.Data["AC_ID"])
+			templateParameters["ACSecret"] = string(secret.Data["AC_SECRET"])
+			Log.Info("Using ApplicationCredentials auth")
+		}
```

Comment:
How do we handle the case where we are unable to retrieve the ACSecret?  Looks like we simply continue and use password auth instead.  But maybe we should requeue?  Or at the very least log the error?